### ISSUE TITLE
Walk headlines should not export just level 1 headlines

### ISSUE THREAD

**Issue by kaushalmodi** [kaushalmodi/ox-hugo#21]
If I have

```org
* Section A
** Section A Post 1
** Section A Post 2
* Section B
** Section B Post 1
** Section B Post 2
```
it should export one `.md` file for each "Post" heading.

*Need to now figure out how to mark a subtree as a "Post" subtree.*

**Comment by titaniumbones** [comment]
Agreed.  I guess the choices are tags and properties, right? I tend to use properties myself but that is just habit.

**Comment by kaushalmodi** [comment]
I like properties too. Want to make the `HUGO_DATE` property mandatory?  A sub-tree having that property wkjld be counted as a single post. 

On a related note, we'll need helper functions to add or update that property only in the current sub-tree.

**Comment by titaniumbones** [comment]
I htink `HUGO_DATE` makes perfect sense for blogs, but maybe is less obvious for other kinds of sites.  Is "EXPORT_TITLE" too general?  

Was away from home for a few days but am back and would like to try to contribute to this project again, though I'm a little busy so might be less active.

**Comment by kaushalmodi** [comment]
> I think HUGO_DATE makes perfect sense for blogs, but maybe is less obvious for other kinds of sites. Is "EXPORT_TITLE" too general?

I was planning on simply taking the [default front matter variables][1], upcasing them and prefixing those with `HUGO_`.

So:

- `title` -> `HUGO_TITLE`
- `date` -> `HUGO_DATE`

and so on..

[1]: https://gohugo.io/content/front-matter/

**Comment by kaushalmodi** [comment]
> Was away from home for a few days but am back and would like to try to contribute to this project again, though I'm a little busy so might be less active.

Same here :) Will be back at full speed soon.

**Comment by titaniumbones** [comment]
> I was planning on simply taking the default front matter variables, upcasing them and prefixing those with HUGO_.

That's what I started with too. But now I wonder whether we should avoid "hugo-izing" as much as we can, and use the generic variables where possible. That makes it easier for users to switch back and forth between platforms.

**Comment by kaushalmodi** [comment]
> But now I wonder whether we should avoid "hugo-izing" as much as we can, and use the generic variables where possible.

Well, it's an exporter for Hugo. So here are my thoughts:

- We should not 'hugo'ize the content. And thus the reluctance to use short codes in content.
- But we do need to convey the meta data required for `hugo` to work.. and thus, if you say, 'hugo'izing stuff just for Hugo front matter may be fine?

I just committed this ( https://github.com/kaushalmodi/ox-hugo/commit/093bdeab2d326fb1c86919cc31336bc67ab7f989 ) before I read your last message. 

May be it is completely fine to get rid of the "HUGO_" prefix completely.. let's see.

Once the this issue ( https://github.com/kaushalmodi/ox-hugo/issues/22 ) is resolved, we will have better stability with front matter spec.

**Comment by kaushalmodi** [comment]
@titaniumbones Thinking more, it might be better to just get rid of the "HUGO_" prefix, else the property drawer keys will get too wordy (example: "EXPORT_HUGO_CATEGORIES"). 

See my last commit ( https://github.com/kaushalmodi/ox-hugo/commit/c52677bf3ef1ff53160676ab7edee0fbfd25a092 ).

From your earlier post:

> Is "EXPORT_TITLE" too general?

Specific to that, was thinking of **not** supporting "*_TITLE" altogether..

- If user is writing one post per Org file, `#+TITLE:`  will be used as Hugo post "title".
- If user is writing one post per subtree (multiple posts per Org file), the subtree heading will be used as Hugo post "title".

Thoughts?

**Comment by kaushalmodi** [comment]
This is sort of implemented in https://github.com/kaushalmodi/ox-hugo/commit/e8566709b15257357b40e6899975b1058e08520c with a new interactive function `org-hugo-export-current-subtree`.  That will export only the current or parent subtree which first matches the condition that an EXPORT_FILE_NAME property be present.

**Comment by kaushalmodi** [comment]
This is now fixed in https://github.com/kaushalmodi/ox-hugo/commit/e026a74ad9e8440659a8b07d3c610ae814e5bf62
